### ISSUE TITLE
Add label_smoothing to SparseCategoricalCrossentropy

### ISSUE THREAD

**Issue by Neilblaze** [keras-team/keras#23487]
## Summary

`label_smoothing` is supported on four of the five crossentropy losses in Keras, but not on `SparseCategoricalCrossentropy`.

| Loss | `label_smoothing` |
| --- | --- |
| `BinaryCrossentropy` | yes |
| `BinaryFocalCrossentropy` | yes |
| `CategoricalCrossentropy` | yes |
| `CategoricalFocalCrossentropy` | yes |
| `SparseCategoricalCrossentropy` | no |

## Motivation

Label smoothing is a standard part of most modern classification and sequence to sequence recipes. The original Transformer paper trains with `0.1`, and the same setting shows up across Inception v3, ViT and the common timm recipes. PyTorch exposes it directly on `nn.CrossEntropyLoss`, which takes integer targets, so anyone porting one of those recipes over expects the sparse loss in Keras to accept it as well.

Today the only way to smooth integer labels is to one-hot encode them first and switch to `CategoricalCrossentropy`. That materializes a target the same shape as `y_pred`. For a segmentation map, or a language model over a large vocabulary, that tensor is already the dominant activation, so doubling it is a real cost for what is conceptually just a different target.

The one-hot target is not actually needed. The smoothed target is

```
(1 - e) * one_hot(y_true) + e / num_classes
```

so the loss splits into the ordinary hard label term plus `e` times the crossentropy against a uniform target. Both pieces can be computed straight from the integer label.

## Proposed API

```python
keras.losses.SparseCategoricalCrossentropy(
    from_logits=False,
    ignore_class=None,
    reduction="sum_over_batch_size",
    axis=-1,
    label_smoothing=0.0,
    name="sparse_categorical_crossentropy",
    dtype=None,
)

keras.losses.sparse_categorical_crossentropy(
    y_true,
    y_pred,
    from_logits=False,
    ignore_class=None,
    axis=-1,
    label_smoothing=0.0,
)
```

The new argument sits after `axis` rather than after `from_logits`, where the other crossentropy losses put it, so that an existing positional call such as `sparse_categorical_crossentropy(y, p, True, 255)` keeps its current meaning. Happy to move it if you would rather the ordering matched the siblings.

The result would match `categorical_crossentropy` on the equivalent one-hot label, and `label_smoothing=0.0` would leave the current output untouched.

I have this working locally and would like to send a PR if the change makes sense.

**Comment by Neilblaze** [comment]
CC'ing: @jeffcarp for greater visibility. Thanks! :)

**Comment by Neilblaze** [comment]
**UPDATE:** A draft of the PR can be found here https://github.com/keras-team/keras/pull/23488

**Comment by jeffcarp** [comment]
Thanks! Will take a look. Do we want to add this to `SparseCategoricalFocalCrossentropy` as well? (cc #23442)
